src/commands/test/lib/nunit-xml-util.ts

```diff
@@ -106,28 +129,51 @@ export class NUnitXmlUtil extends XmlUtil {
   }
 
   combineTestResultsAttribute(xml1: Document, xml2: Document, attributeName: string) {
-    this.addTestResultsAttribute(xml1, attributeName, this.getTestResultsAttribute(xml2, attributeName));
+    this.addResultsAttribute(xml1, attributeName, this.getResultsAttribute(xml2, attributeName));
   }
 
-  getTestResultsAttribute(xml: Document, attributeName: string): number {
-    const testResults: Element[] = this.collectAllElements(xml.documentElement, "test-results");
-    if (testResults.length === 0) {
+  getResultsAttribute(xml: Document, attributeName: string): number {
+    const testResults: Element = this.getResultsNode(xml);
+    if (!testResults) {
       return 0;
     }
 
-    const attr: Attr = testResults[0].attributes.getNamedItem(attributeName);
+    const attr: Attr = testResults.attributes.getNamedItem(attributeName);
     if (attr.value) {
       return Number(attr.value);
     }
     return 0;
   }
 
-  addTestResultsAttribute(xml: Document, attributeName: string, value: number) {
-    const currentValue: number = this.getTestResultsAttribute(xml, attributeName);
-    const testResults: Element[] = this.collectAllElements(xml.documentElement, "test-results");
-    const attr: Attr = testResults[0].attributes.getNamedItem(attributeName);
+  addResultsAttribute(xml: Document, attributeName: string, value: number) {
+    const currentValue: number = this.getResultsAttribute(xml, attributeName);
+    const testResults: Element = this.getResultsNode(xml);
+
+    if (!testResults) {
+      return;
+    }
+
+    const attr: Attr = testResults.attributes.getNamedItem(attributeName);
     if (attr) {
       attr.value = String(currentValue + value);
     }
   }
+
+  getResultsNode(xml: Document): Element {
+    let testResults: Element[] = this.collectAllElements(xml.documentElement, "test-results");
+    if (testResults.length === 0) {
+      testResults = this.collectAllElements(xml.documentElement, "test-run");
+    }
+
+    return testResults[0];
+  }
+
+  isNUnit3(xml: Document): boolean {
+    const testResults: Element[] = this.collectAllElements(xml.documentElement, "test-results");
```

Comment:
this is more of a not nunit2 test. Perhaps its better to test for precense of "test-run" instead of absence of "test-results"

Reply:
I've done it like this pending NUnit4 coming out so that it's easier to convert to:
> getNUnitVersion: int
Or something similar. Although it might not be the case, I figured NUnit4 is more likely to follow similar report format the NUnit3 - at which point we may need to dig in to the version attribute on the `test-results` node.